### DISCUSSION TITLE
Transition Delay Effect

### DISCUSSION THREAD

**Discussion by marcelo-3k** [tailwindlabs/tailwindcss#12005] [Help]
Tailwind CSS Version: v3.3.3
Framework: Next.js
Browser: Google Chrome
OS: Windows
#
### Problem
I'm trying to create an increasing effect through opacity using an increasing delay to cause such an effect, but without success, it's as if the classes don't respond.

I also tried adding/extending the js configurations, but without success.

So first I'm trying to deal with arbitrary values
#
### How to reproduce

```tsx
function NavLinks({ isMenuOpen }: NavLinksProps) {
  const pathname = usePathname()

  const navlinks = [
    {
      href: '/projects',
      name: 'Projetos'
    },
    {
      href: '/services',
      name: 'Serviços'
    },
    {
      href: '/about',
      name: 'Sobre'
    },
    {
      href: '/contact',
      name: 'Contato'
    },
  ]

  return (
    <ul
      className={
        cn(
          "w-full h-auto pt-28 pb-12 px-5 flex flex-col justify-start gap-2 text-lg text-brand-title bg-brand-primary absolute left-0 z-40 transition-all duration-500 ease-in-out md:w-fit md:py-0 md:static md:items-center md:gap-16 md:flex-row md:bg-transparent",
          isMenuOpen ? 'top-0' : '-top-full delay-500'
        )
      }
    >
      {navlinks.map((link, index) => {
        const isActive = pathname === link.href

        const delay = 500 + (index * 50)
        const decrescentDelay = (navlinks.length - index) * 50 + 100
        // console.log('asc-transition-delay: ', delay)
        // console.log('desc-transition-delay: ', decrescentDelay)
        return (
          <li
            key={link.name}
            className={
              cn(
                "px-0 py-4 text-inherit md:text-brand-title md:text-lg transition-opacity md:opacity-100 md:w-fit",
                isMenuOpen ?  `delay-[${delay}ms]` : `delay-[${decrescentDelay}ms]`
              )
            }
          >
            <Link
              href={link.href}
              className={
                cn(
                  "block w-full",
                  isActive ? "underline underline-offset-4 decoration-brand-title" : "no-underline"
                )
              }
            >
              {link.name}
            </Link>
          </li>
        )
      })}
    </ul >
  )
}
```

**Comment by wongjn** [comment]
As per [the documentation](https://tailwindcss.com/docs/content-configuration#dynamic-class-names):

> The most important implication of how Tailwind extracts class names is that it will only find classes that exist as _complete unbroken strings_ in your source files.
>
> If you use string interpolation or concatenate partial class names together, Tailwind will not find them and therefore will not generate the corresponding CSS:
>
> **Don’t construct class names dynamically**
>
> ```vue
> <div class="text-{{ error ? 'red' : 'green' }}-600"></div>
> ```
> In the example above, the strings `text-red-600` and `text-green-600` do not exist, so Tailwind will not generate those classes.
> Instead, make sure any class names you’re using exist in full:
>
> **Always use complete class names**
>
> ```vue
> <div class="{{ error ? 'text-red-600' : 'text-green-600' }}"></div>
> ```

You could consider using the `style` attribute instead like:

```jsx
<li
  key={link.name}
  className={cn("…")}
  style={{ delay: `${isMenuOpen ? delay : decrescentDelay}ms` }}
>
```

**Comment by marcelo-3k** [comment]
> As per [the documentation](https://tailwindcss.com/docs/content-configuration#dynamic-class-names):
> 
> > The most important implication of how Tailwind extracts class names is that it will only find classes that exist as _complete unbroken strings_ in your source files.
> > If you use string interpolation or concatenate partial class names together, Tailwind will not find them and therefore will not generate the corresponding CSS:
> > **Don’t construct class names dynamically**
> > ```
> > <div class="text-{{ error ? 'red' : 'green' }}-600"></div>
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > In the example above, the strings `text-red-600` and `text-green-600` do not exist, so Tailwind will not generate those classes.
> > Instead, make sure any class names you’re using exist in full:
> > **Always use complete class names**
> > ```
> > <div class="{{ error ? 'text-red-600' : 'text-green-600' }}"></div>
> > ```
> 
> You could consider using the `style` attribute instead like:
> 
> ```js
> <li
>   key={link.name}
>   className={cn("…")}
>   style={{ delay: `${isMenuOpen ? delay : decrescentDelay}ms` }}
> >
> ```

Yeah, with style prop works fine, but in case of class not even the HTML returns a correct classes in DOM but effect is not apply.

console elements: 
![image](https://github.com/tailwindlabs/tailwindcss/assets/138832209/07cbefe8-3c26-4242-9d3e-57cdd14e2e37)

**Reply by wongjn** [reply]
Yes, as previously stated, you would need to use the `style` attribute. As per [the documentation](https://tailwindcss.com/docs/content-configuration#dynamic-class-names):

> The most important implication of how Tailwind extracts class names is that it will only find classes that exist as _complete unbroken strings_ in your source files.
>
> If you use string interpolation or concatenate partial class names together, Tailwind will not find them and therefore will not generate the corresponding CSS:
>
> **Don’t construct class names dynamically**
>
> ```vue
> <div class="text-{{ error ? 'red' : 'green' }}-600"></div>
> ```
> In the example above, the strings `text-red-600` and `text-green-600` do not exist, so Tailwind will not generate those classes.
> Instead, make sure any class names you’re using exist in full:
>
> **Always use complete class names**
>
> ```vue
> <div class="{{ error ? 'text-red-600' : 'text-green-600' }}"></div>
> ```

**Reply by marcelo-3k** [reply]
its a solution but sad =)
Thanks
